node/tests/integration/main.rs

```diff
@@ -418,7 +425,483 @@ async fn test_play_check() {
     execute_tx_and_sync(&mut client, tx_request).await;
 
     println!("Executed and synced with node");
-    assert_slot_status_check(&client, target_account_id, game_slot_data.clone()).await;
+    assert_slot_status_check(&client, target_account_id, game_slot_data.clone(), 1 as u8).await;
+}
+
+#[tokio::test]
+async fn test_e2e() {
+    let mut client: AzeClient = create_test_client();
+
+    let (game_account, player1_account_id, faucet_account_id, game_slot_data) = setup_accounts(
+        &mut client
+    );
+
+    let game_account_id = game_account.id();
+
+    let (player2_account, _) = client
+        .new_game_account(
+            AzeAccountTemplate::PlayerAccount {
+                mutable_code: false,
+                storage_mode: AccountStorageMode::Local,
+            },
+            None
+        )
+        .unwrap();
+    
+    let (player3_account, _) = client
+        .new_game_account(
+            AzeAccountTemplate::PlayerAccount {
+                mutable_code: false,
+                storage_mode: AccountStorageMode::Local,
+            },
+            None
+        )
+        .unwrap();
+
+    let (player4_account, _) = client
+        .new_game_account(
+            AzeAccountTemplate::PlayerAccount {
+                mutable_code: false,
+                storage_mode: AccountStorageMode::Local,
+            },
+            None
+        )
+        .unwrap();
+
+    // Preflop
+
+    // Player 1 --> Small blind bets SMALL_BLIND_AMOUNT
+    let player1_bet = SMALL_BLIND_AMOUNT;
+    bet(&mut client, player1_account_id, game_account_id, faucet_account_id, player1_bet, 1 as u8).await;
+    //check player balance
+    let game_account = client.get_account(game_account_id).unwrap().0;
+    assert_eq!(
+        game_account.storage().get_item(68 as u8),
+        RpoDigest::new([Felt::from((PLAYER_INITIAL_BALANCE - player1_bet) as u8), Felt::ZERO, Felt::ZERO, Felt::ZERO])
+    );
+    println!("----->>> Small blind betted");
+
+    // Player 2 --> Big blind bets SMALL_BLIND_AMOUNT * 2
+    let player2_bet = SMALL_BLIND_AMOUNT * 2;
+    bet(&mut client, player2_account.id(), game_account_id, faucet_account_id, player2_bet, 2 as u8).await;
+    // check balance
+    let game_account = client.get_account(game_account_id).unwrap().0;
+    assert_eq!(
+        game_account.storage().get_item(81 as u8),
+        RpoDigest::new([Felt::from((PLAYER_INITIAL_BALANCE - player2_bet) as u8), Felt::ZERO, Felt::ZERO, Felt::ZERO])
+    );
+    println!("----->>> Big blind betted");
+
+    // Deal cards to players and assert the account status
+    deal_card(&mut client, game_account_id, player1_account_id, faucet_account_id, 0).await;

```

Comment:
Although as per the current game play it doesn't matter when to deal cards, But i would recommend keeping these at the start of the test.

Reply:
Hmm, Looked in. In most of them they are